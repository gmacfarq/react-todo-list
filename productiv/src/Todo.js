import React, {useState} from "react";

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority, completed}
 * - handleComplete
 *
 * { EditableTodo, TopTodo } -> Todo
 **/

function Todo({todo, complete}) {
  const [isCompleted, setIsCompleted] = useState(todo.completed);

  /** toggle the state to render todos crossed out*/
  function toggleCompleted(){
    setIsCompleted(isCompleted => !isCompleted)
  }

  /** calls the complete function from this component*/
  function handleComplete(){
    toggleCompleted()
    complete(todo.id, !isCompleted)
  }

  return (


     <div className="Todo">
        <div>
          <b onClick={handleComplete}>
            {isCompleted ? <s>{todo.title}</s> : todo.title}
          </b>
          <small> (priority: {todo.priority})</small>
        </div>
        <div><small>{todo.description}</small></div>
      </div>
  );
}

export default Todo;
