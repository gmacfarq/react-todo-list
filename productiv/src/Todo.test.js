import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";
import testTodos from "./initTodos";



it("renders Todo  without crashing", function () {
    render(
      <Todo
        todo={testTodos[0]}
  />
  );
  });

it("matches snapshot", function () {
  const { container } = render(
    <Todo
      todo={testTodos[0]}
/>
  );
  expect(container).toMatchSnapshot();
});