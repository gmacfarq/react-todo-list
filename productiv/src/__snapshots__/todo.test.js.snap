// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Code!
      </b>
       
      <small>
        2
      </small>
    </div>
    <div>
      <small>
        Write some code
      </small>
    </div>
  </div>
</div>
`;
